src/components/gui/tabs/query-tab.tsx

```diff
@@ -84,6 +87,31 @@ export default function QueryWindow({
     initialNamespace ?? "Unsaved Query"
   );
   const [savedKey, setSavedKey] = useState<string | undefined>(initialSavedKey);
+  const [placeHolders, setPlaceHolders] = useState<Record<string, string>>({});
+
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      const editorState = editorRef.current?.view?.state;
+      if (!editorState) return;
+      const finalStatements = splitSqlQuery(editorState).map((q) => q.text);
```

Comment:
Why do we need to split the sql to find placeholder? I think `tokenizeSql` should be able to tokenized the whole sql and find the placeholder